Modules/socketmodule.c

```diff
@@ -5574,6 +5574,23 @@ static PyType_Spec sock_spec = {
     .slots = sock_slots,
 };
 
+static PyObject *
+socket_getattr(PyObject *self, PyObject *name)
+{
+    const char *attrname = PyUnicode_AsUTF8(name);
+    PyObject *sock_type = PyType_FromSpec(&sock_spec);
+    if (sock_type == NULL) {
+        return NULL;
+    }
+
+    if (!strcmp(attrname, "SocketType")) {
+        PyErr_Warn(PyExc_DeprecationWarning, "Use socket.socket type instead");
```

Comment:
I'd advise using the same message as for the Python module. In addition, shouldn't we mention when it'll be removed (because the Python implementation mentions it).